crates/ty_python_semantic/resources/mdtest/annotations/self.md

```diff
@@ -153,18 +156,45 @@ class B:
         reveal_type(self)  # revealed: Self@keyword_only
         return self
 
+    @some_decorator
+    def decorated_method(self) -> Self:
+        reveal_type(self)  # revealed: Self@decorated_method
+        return self
+
     @property
     def a_property(self) -> Self:
-        # TODO: Should reveal Self@a_property
-        reveal_type(self)  # revealed: Unknown
+        reveal_type(self)  # revealed: Self@a_property
         return self
 
+    async def async_method(self) -> Self:
+        reveal_type(self)  # revealed: Self@async_method
+        return self
+
+    @staticmethod
+    def static_method(self):
+        # The parameter can be called `self`, but it is not treated as `Self`
+        reveal_type(self)  # revealed: Unknown
+
+    @staticmethod
+    @some_decorator
+    def decorated_static_method(self):
+        reveal_type(self)  # revealed: Unknown
+
+    @some_decorator
+    @staticmethod
```

Comment:
we should actually reject this on Python <3.10, because instances of `staticmethod` are not callable on Python <3.10, and `some_decorator` states that it only accepts `Callable`s. Worth a TODO comment?